src/meltano/core/tracking/tracker.py

```diff
@@ -125,7 +125,11 @@ def __init__(  # - too many local variables, too many expressions
         emitters: list[Emitter] = []
         for endpoint in endpoints:
             if not check_url(endpoint):
-                logger.warning("invalid_snowplow_endpoint", endpoint=endpoint)
+                warn(
```

Comment:
This does seem to get rid of the test flakiness, but feels a bit heavy-handed to me.

Here's how it looks like in the console:

```shell
2025-01-22T15:38:18.718632Z [warning  ] /projects/meltano/src/meltano/cli/cli.py:143: UserWarning: Invalid Snowplow endpoint: notvalid:8080
  ctx.obj["tracker"] = Tracker(project)

2025-01-22T15:38:18.719858Z [info     ] The default environment 'dev' will be ignored for `meltano select`. To configure a specific environment, please use the option `--environment=<environment name>`.
2025-01-22T15:38:18.777064Z [warning  ] /projects/meltano/src/meltano/core/plugin_invoker.py:178: UserWarning: Invalid Snowplow endpoint: notvalid:8080
  self.tracker = Tracker(project)
```